core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1525,6 +1537,9 @@ private void deleteFromEntry(DeleteFileEntry entry) {
         // Do not journal entries covered recursively for performance
         if (delInode.getId() == inode.getId() || unsafeInodes.contains(delInode.getParentId())) {
           mInodeTree.deleteInode(tempInodePath, opTimeMs, deleteOptions, journalContext);
+          if (delInode.isFile()) {
+            blockDeletionContext.registerBlocksForDeletion(((InodeFile) delInode).getBlockIds());
+          }
```

Comment:
Should this logic be in `deleteInode`?

Reply:
yes, this was actually necessary to cover the in-sync recursive delete case. I've added an integration test and moved the logic into `deleteInode`